subsys/shell/shell.c

```diff
@@ -1371,6 +1378,8 @@ int shell_init(const struct shell *shell, const void *transport_config,
 	shell->ctx->tid = tid;
 	k_thread_name_set(tid, shell->thread_name);
 
+	shell_ctx = (const struct shell *)(shell->ctx);
```

Comment:
Not sure I understand this.

We have a shell of the type `struct shell`, and then you assign the `shell->ctx` (which is a `struct shell_ctx` to a variable `shell_ctx`, which has the type `struct shell`. 

Basically you are casting the `shell->ctx` to a new type; doesn't that give all sorts of problems?

Reply:
my bad I misread the code, fixed now